src/DataGrid.tsx

```diff
@@ -432,6 +433,25 @@ function DataGrid<R, SR, K extends Key>(
     }
   });
 
+  useLayoutEffect(() => {
+    if (!isWidthInitialized || flexWidthViewportColumns.length === 0) return;
+    const newColumnWidths = new Map<string, number>();
+    for (const column of flexWidthViewportColumns) {
+      const columnElement = gridRef.current!.querySelector<HTMLDivElement>(
+        `[aria-colindex="${column.idx + 1}"]`
+      );
+      if (columnElement) {
```

Comment:
`flexWidthViewportColumns` only contains rendered columns, so is this check necessary?
Can `newColumnWidths` ever be empty after this loop?

Reply:
IIRC, it was failing in some cases. It is possible to have `flexWidthViewportColumns .length > renderedFlexWidthViewportColumns.length`

Reply:
>It is possible to have flexWidthViewportColumns .length > renderedFlexWidthViewportColumns.length

Is it? `flexWidthViewportColumns` is a subset of `viewportColumns` though

Reply:
This can fail if header cell has a colSpan